crates/sparrow-runtime/src/stores/object_stores.rs

```diff
@@ -23,29 +21,27 @@ use super::{object_store_url::ObjectStoreKey, ObjectStoreUrl};
 /// For now, the registry exists as a cache for the clients due to the overhead
 /// required to create the cache. The future goal for the registry is to
 /// control the number of possibile open connections.
-#[derive(Default)]
+#[derive(Default, Debug)]
 pub struct ObjectStoreRegistry {
-    object_stores: RwLock<HashMap<ObjectStoreKey, Arc<dyn ObjectStore>>>,
+    object_stores: DashMap<ObjectStoreKey, Arc<dyn ObjectStore>>,
```

Comment:
What's the reasoning to switch from RwLock -> DashMap? Do we get performance/simplification?

Reply:
Generally just simplification (as shown from all the deletions). There are performance benefits but I don't think they're compelling. Basically -- RwLock forces us to manage the lock, and lock the entire hashmap. DashMap does that internally and manages the lock under the hood.